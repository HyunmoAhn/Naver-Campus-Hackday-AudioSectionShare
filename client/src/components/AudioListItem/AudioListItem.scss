.AudioListItem {
  border-bottom: 2px solid green;
  padding: 2px;
  text-align: center;
  overflow: hidden;
}

.AudioListItem__title {
  width: 90%;
  float: left;
  white-space: nowrap;
  overflow: scroll;
}

.AudioListItem__count {
  width: 10%;
  float: right;
  border-left: 3px solid green;
}