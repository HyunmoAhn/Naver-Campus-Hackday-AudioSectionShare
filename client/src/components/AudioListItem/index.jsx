import React from 'react';
import PropTypes from 'prop-types';
import './AudioListItem.scss';

const propTypes = {
	item: PropTypes.object.isRequired,
	router: PropTypes.object,
	onListInfo: PropTypes.func,
};

class AudioListItem extends React.Component {
	render() {
		const { count, id, title } = this.props.item;
		return (
			<div
				className="AudioListItem"
			  onClick={() => {
					this.props.onListInfo(id);
				  this.props.router.push(`?id=${id}&isList=true`);
			  }}
			>
				<span className="AudioListItem__title">
					{title}
				</span>
				<span className="AudioListItem__count">
					{count}
				</span>
			</div>
		)
	}
}

AudioListItem.propTypes = propTypes;

export default AudioListItem;