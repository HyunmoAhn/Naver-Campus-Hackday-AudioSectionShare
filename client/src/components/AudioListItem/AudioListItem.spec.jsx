import React from 'react';
import { shallow } from 'enzyme';

import AudioListItem from './';

describe('<AudioListItem />', () => {
	const mockItem = {
		title: 'mock-title',
		id: 'mock-id',
		count: 5,
	};

	it('should match to snapshot when render default', () => {
		const wrapper = shallow(<AudioListItem item={mockItem} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should call onListInfo and router.push when click .AudioListItem', () => {
		const onListInfo = jest.fn();
		const mockRouter = {
			push: () => false,
		};
		const wrapper = shallow(<AudioListItem
			item={mockItem}
			router={mockRouter}
			onListInfo={onListInfo}
		/>);
		const routerPush = jest.spyOn(mockRouter, 'push');
		const listItem = wrapper.find('.AudioListItem');

		listItem.simulate('click');
		expect(onListInfo.mock.calls.length).toBe(1);
		expect(onListInfo.mock.calls[0]).toEqual([wrapper.instance().props.item.id]);
		expect(routerPush.mock.calls.length).toBe(1);
		expect(routerPush.mock.calls[length])
			.toEqual([`?id=${wrapper.instance().props.item.id}&isList=true`]);
	});
});