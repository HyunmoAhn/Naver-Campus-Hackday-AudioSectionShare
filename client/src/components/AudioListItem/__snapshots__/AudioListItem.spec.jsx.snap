// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AudioListItem /> should match to snapshot when render default 1`] = `
<div
  className="AudioListItem"
  onClick={[Function]}
>
  <span
    className="AudioListItem__title"
  >
    mock-title
  </span>
  <span
    className="AudioListItem__count"
  >
    5
  </span>
</div>
`;
